base_rest_demo/__manifest__.py

```diff
@@ -18,10 +18,8 @@
         "component",
```

Comment:
The file **naive_orm_model.py** inside the **pydnatic_models** folder, import utils from odoo.addons.pydantic. But pydantic (odoo-addon-pydnatic) is not specified inside the dependance. There is a reason for that?

Reply:
No, module dependencies are known to work transitively. You're the first to notice in 4,5 years ;-) https://github.com/OCA/rest-framework/blob/b0c3cd48044a1957ef6fe847c4c206a530789b2a/base_rest_demo/__manifest__.py. I'll remove it if I need to make other changes as well.

Reply:
I think it would be better to add it in the dependencies, because if somebody install only base_rest_demo to test the module for any reason. He will get a python error at runtime because odoo-addon-pydantic will not be installed.

Reply:
I'm sorry, I misunderstood. My mind is strange when I encounter a typo. I literally did not recognize `pydnatic` as an existing module so I backtracked to see what else could be wrong so my focus was on removing the dependency on `component` which is strictly speaking redundant.

As for the 'missing' dependency on `pydantic`, this is not an error because `base_rest_demo` depends on `base_rest_pydantic` which in turn depends on `pydantic`: https://github.com/OCA/rest-framework/pull/313/files#diff-319e5c838c0a36b0b147dd7c52b013a6ddec30fbc45d85292f4469001ab35a45R16.

Reply:
First, sorry for the typo.
Secondly, we have to be careful because there are two pydantic dependencies. The first one  is the external python dependancy to `pydantic` that you explained well. The second one if a odoo module named `pydantic` that you also migrated in this pull request.  https://github.com/OCA/rest-framework/pull/313/files#diff-469a545c098c2ceb85d04dbcce595ba37689209f8e687971eb5ba8b93061c26b

I don't know how to show the line, but this is a link to the line that disturb me:
https://github.com/OCA/rest-framework/blob/780fb402de43f571206f7c302b14f2e21bf52468/base_rest_demo/pydantic_models/naive_orm_model.py#L6

I hope I expressed myself better. 

Sorry again @StefanRijnhart

Reply:
@FrancoMaxime Good catch. Indeed, `pydancic` is missing into the depends of base_rest_demo. This should be fixed in 14.0 and forward ported...

Reply:
Thanks for clarifying @FrancoMaxime! Would you like to propose this for 14 and 15, and then I'll pick it into this one.

Reply:
Hello @lmignon and @StefanRijnhart I made two pull request for the fix in 14.0  and the forward port to 15.0 for this discussion. I hope I made good comments on these requests.
https://github.com/OCA/rest-framework/pull/328
https://github.com/OCA/rest-framework/pull/329

Reply:
Thank you @FrancoMaxime #328 is merged and #329 will be once CI will be functional again

Reply:
Thank you @FrancoMaxime. Picked into this one now.